functions/Copy-DbaResourceGovernor.ps1

```diff
@@ -109,26 +113,93 @@ function Copy-DbaResourceGovernor {
             Notes             = $null
             DateTime          = [DbaDateTime](Get-Date)
         }
+        
+        $copyResourceGovClassifierFunc = [pscustomobject]@{
+            SourceServer      = $sourceServer.Name
+            DestinationServer = $destServer.Name
+            Type              = "Resource Governor Settings"
+            Name              = "Classifier Function"
+            Status            = $null
+            Notes             = $null
+            DateTime          = [DbaDateTime](Get-Date)
+        }
 
         if ($Pscmdlet.ShouldProcess($destination, "Updating Resource Governor settings")) {
             if ($destServer.Edition -notmatch 'Enterprise' -and $destServer.Edition -notmatch 'Datacenter' -and $destServer.Edition -notmatch 'Developer') {
                 Write-Message -Level Verbose -Message "The resource governor is not available in this edition of SQL Server. You can manipulate resource governor metadata but you will not be able to apply resource governor configuration. Only Enterprise edition of SQL Server supports resource governor."
             }
             else {
                 try {
-                    $sql = $sourceServer.ResourceGovernor.Script() | Out-String
-                    Write-Message -Level Debug -Message $sql
-                    Write-Message -Level Verbose -Message "Updating Resource Governor settings."
-                    $destServer.Query($sql)
+                    Write-Message -Level Verbose -Message "Managing classifier function."
+                    if (!$sourceClassifierFunction) {
+                        $copyResourceGovClassifierFunc.Status = "Skipped"
+                        $copyResourceGovClassifierFunc.Notes = $null
+                        $copyResourceGovClassifierFunc | Select-DefaultView -Property DateTime, SourceServer, DestinationServer, Name, Type, Status, Notes -TypeName MigrationObject
+                    }
+                    else {
+                        $fullyQualifiedFunctionName = $sourceClassifierFunction.Schema+"."+$sourceClassifierFunction.Name
+                                  
+                        if (!$destClassifierFunction) {
+                            $destFunction = $destServer.Databases["master"].UserDefinedFunctions[$sourceClassifierFunction.Name]
+
+                            if ($destFunction) {
+                                Write-Message -Level Verbose -Message "Dropping the function with the source classifier function name."
```

Comment:
Anything where you are modifying the object (or dropping it) needs to be wrapped in a ShouldProcess block. Not sure if the other copy commands do one blanket block but since we are adding the features for pulling the functions it would be good to add the whatif support as well.